core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -361,6 +383,54 @@ public void close() throws IOException {
     httpClient.close(CloseMode.GRACEFUL);
   }
 
+  public static HTTPClient buildFrom(Map<String, String> properties) {
```

Comment:
Switching from default builder pattern to property initialization pattern so that we can expose configurations in the future for things like pool sizes, request/response timeouts, etc. via properties as opposed to having to replicate the apache http client builder.